src/blob/file.rs

```diff
@@ -255,18 +268,34 @@ impl File {
             ioring,
             no_lock_fd: Arc::new(std_file),
             size,
+            dirty_bytes: Arc::new(AtomicUsize::new(0)),
         };
         Ok(file)
     }
 
     pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
         if let Some(ref ioring) = self.ioring {
             let compl = ioring.fsync(&*self.no_lock_fd);
-            compl.await
+            compl.await?
         } else {
             let fd = self.no_lock_fd.clone();
-            Self::background_sync_call(move || fd.sync_all()).await
-        }
+            Self::background_sync_call(move || fd.sync_all()).await?
+        };
+        self.dirty_bytes.store(0, ORD);
+        Ok(())
+    }
+
+    pub(crate) fn background_fsyncdata(&self) {
+        let fd = self.no_lock_fd.clone();
+        if let Some(ref ioring) = self.ioring {
+            let ioring = ioring.clone();
+            tokio::spawn(async move {
```

Comment:
We should track the spawning task somewhere. The best place for this is `ObserverWorker`